app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -1,87 +1,103 @@
+
+const PASSWORD: PasswordField = 'user[password]'
+const PASSWORD_CONFIRMATION: PasswordConfirmationField = 'user[password_confirmation]'
 
-const PASSWORD = 'user[password]'
-const PASSWORD_CONFIRMATION = 'user[password_confirmation]'
-const validationConstraints = {
+// The following validation objects are tied to FormGroups helperTexts, very closed module, not open to change. TODO: REFACTOR those components
+const validationConstraints: Constraints = {
   [PASSWORD]: {
-    presence: true,
-    length: { minimum: 1 }
+    presence: {
+      message: '^isMandatory'
+    },
+    length: {
+      minimum: 1,
+      message: '^password'
+    }
   },
   [PASSWORD_CONFIRMATION]: {
-    presence: true,
-    length: { minimum: 1 },
-    equality: PASSWORD
+    presence: {
+      message: '^isMandatory'
+    },
+    length: {
+      minimum: 1,
+      message: '^isMandatory'
+    },
+    equality: {
+      attribute: PASSWORD,
+      message: '^mustMatch'
+    }
   }
 }
 
-const isFormValid = (formNode) => validateForm(formNode, validationConstraints)
-
-const getPasswordConfirmationError = (errors) => {
-  const passwordConfirmationError = (errors && errors[PASSWORD_CONFIRMATION]) && errors[PASSWORD_CONFIRMATION][0]
-  const passwordsMatchError = passwordConfirmationError && passwordConfirmationError.includes('is not equal to')
-  return {
-    passwordConfirmationError: !!passwordConfirmationError,
-    errorMessage: passwordsMatchError ? 'mustMatch' : 'isMandatory'
-  }
+const fieldsTemplate: FieldState = {
+  [PASSWORD]: '',
+  [PASSWORD_CONFIRMATION]: ''
 }
 
-const comparePasswordsLength = (event) => {
-  const passwordInput = event.currentTarget['user_password']
-  const passwordConfirmationInput = event.currentTarget['user_password_confirmation']
-  return passwordConfirmationInput.value.length >= passwordInput.value.length
+const fieldErrorsTemplate: FieldErrorsState = {
+  [PASSWORD]: [''],
+  [PASSWORD_CONFIRMATION]: ['']
 }
 
+const extractFirstErrorMessage = (errorMessageArray: string[]): string => errorMessageArray.slice(0, 1).pop()
+const isFieldValid = (errorMessageArray: string[]): boolean => !errorMessageArray.slice().pop().length
+
 const useFormState = () => {
+  const [fieldValues, setFieldValues] = useState(fieldsTemplate)
+  const [fieldErrors, setFieldErrors] = useState(fieldErrorsTemplate)
   const [isFormDisabled, setIsFormDisabled] = useState(true)
-  const [passwordValue, setPasswordValue] = useState('')
-  const [isPasswordValid, setIsPasswordValid] = useState(undefined)
-  const [passwordConfirmationValue, setPasswordConfirmationValue] = useState('')
-  const [isPasswordConfirmationValid, setIsPasswordConfirmationValid] = useState(undefined)
-  const [confirmationErrorMessage, setConfirmationErrorMessage] = useState('isMandatory')
-  const [validationStarted, setValidationStarted] = useState(false)
+  const [areFieldsPristine, setAreFieldsPristine] = useState({[PASSWORD]: true, [PASSWORD_CONFIRMATION]: true})
```

Comment:
Obviously not a big deal but I wonder why this state doesn't deserver a "template" like the others!
```suggestion
  const pristineTemplate = {
    [PASSWORD]: true,
    [PASSWORD_CONFIRMATION]: true
  }
  const [areFieldsPristine, setAreFieldsPristine] = useState(pristineTemplate)
```